kalite/distributed/static/js/distributed/topics/views.js

```diff
@@ -145,14 +187,40 @@ window.SidebarView = BaseView.extend({
 
     update_sidebar_visibility: function() {
         if (this.state_model.get("open")) {
+            // Used to get left value in number form
+            var sidebarPanelPosition = this.sidebar.position();
             this.sidebar.css({left: 0});
-            this.$(".sidebar-tab").css({left: this.$(".sidebar-panel").width()}).html('<span class="icon-circle-left"></span>');
+            this.resize_sidebar();
+            this.sidebarTab.css({left: this.sidebar.width() + sidebarPanelPosition.left}).html('<span class="icon-circle-left"></span>');
             this.$(".fade").show();
-        } else {
+        } 
+        else {
             this.sidebar.css({left: - this.width});
-            this.$(".sidebar-tab").css({left: 0}).html('<span class="icon-circle-right"></span>');
+            this.sidebarTab.css({left: 0}).html('<span class="icon-circle-right"></span>');
             this.$(".fade").hide();
         }
+
+        this.set_sidebar_back();
```

Comment:
I should make it clearer. `set_sidebar_back()` actually makes sure the `.sidebar-back` button is in whatever state it is supposed to be in. I should call it `set_sidebar_back_btn()` instead. I just decided to go with the shortened version.